concordium-consensus/src/Concordium/GlobalState/BlockState.hs

```diff
@@ -379,6 +388,49 @@ class (Monad m, BlockStateTypes m) => ModuleQuery m where
     -- |Get a module artifact from an 'InstrumentedModuleRef'.
     getModuleArtifact :: Wasm.IsWasmVersion v => InstrumentedModuleRef m v -> m (GSWasm.InstrumentedModuleV v)
 
+-- |We create a wrapper here so we can
+-- derive another 'HashableTo' instance which omits
+-- the exact 'RejectReason' in the resulting hash.
+newtype TransactionSummaryV1 = TransactionSummaryV1 {_transactionSummaryV1 :: TransactionSummary' ValidResult}
+    deriving(Eq, Show)
+
+-- |A 'HashableTo' instance for a 'TransactionSummary'' which
+-- omits the exact reject reason.
+-- Failures are simply tagged with a '0x1' byte.
+-- Note. The hash is computed lazily on purpose as the summary can be large
+-- i.e., the resulting events.
+instance HashableTo H.Hash TransactionSummaryV1 where
+  getHash (TransactionSummaryV1 summary) = H.hashLazy $! S.runPutLazy $!
+      encodeSender (tsSender summary) <>
+      S.put (tsHash summary) <>
+      S.put (tsCost summary) <>
+      S.put (tsEnergyCost summary) <>
+      S.put (tsType summary) <>
+      S.put (tsIndex summary) <>
+      encodeValidResult (tsResult summary)
+    where
+      -- |Encode the 'ValidResult' omitting the exact 'RejectReason' if the
+      -- transaction failed. Otherwise we encode the resulting events in the
+      -- resulting outcome hash.
+      encodeValidResult :: S.Putter ValidResult
+      encodeValidResult (TxSuccess events) = S.putWord8 0 <> putListOf putEvent events
+      -- We omit the exact 'RejectReason'.
+      encodeValidResult (TxReject _) = S.putWord8 1
+      encodeSender :: S.Putter (Maybe AccountAddress)
+      encodeSender Nothing = S.putByteString $! BS.replicate 32 0
+      encodeSender (Just sender) = S.put sender
```

Comment:
While this is almost certainly fine, even better is to also add a tag 0 for nothing, 1 for sender.

Reply:
Good catch! Fixed